src/components/AnonymousChat/BaseChannelItem.tsx

```diff
@@ -39,20 +38,9 @@ const BaseChannelItem: (props: BaseChannelItemProps) => React.ReactElement = ({
   onPress = () => {
     console.log('onPress');
   },
+  showFollowingButton = false,
   dbAnonUserInfo = null
 }) => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `showFollowingButton` prop has been added, while the logic related to the `followContext` and `profileContext` has been removed. This change simplifies the component by moving the responsibility of determining whether a user is followed to the parent component or custom hook. It's important to ensure that the `showFollowingButton` prop is being correctly set in the parent component or hook.
```diff
-  const [profileContext] = (React.useContext(Context) as unknown as any).profile;
-  const [followContext] = (React.useContext(Context) as unknown as any).following;
-  const {myProfile} = profileContext;
-  let isFollowing = false;
-  if (type === BaseChannelItemTypeProps.SIGNED_PM) {
-    const members = postMaker?.members || postMaker?.channel?.members;
-    const targetUser = members?.find((member) => member?.user_id !== myProfile?.user_id)?.user;
-    isFollowing = Boolean(
-      followContext?.users?.find((user) => user?.user_id_followed === targetUser?.id)
-    );
-  }
+  showFollowingButton = false,
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->